### PR TITLE
Draft of public page and functionality implemented

### LINKED ISSUE #15: Implement user's public page
The page contains:
- First and last names
- Public email
- Links

### PR DESCRIPTION
**Pull request by Washubrnfg85**
Only a draft for now.
To do: 

- [x] mechanism to share profile
- [x] something wrong with picture and placeholder - need some time to fix

### CODE REVIEW

**Comment by eugenesmkn** [comment]
The main functionality doesn't work.

Steps to reproduce:
1. Get a completed profile (first/last names, links).
2. Click at "Public page" on navigation menu.
3. Copy the link to share the profile.
4. Open new browser window in **incognito** mode and use the copied link.

Expected result:
Full user profile is displayed on the page.

Actual result:
Redirect to login.

File: src/main/resources/templates/public_page.html [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            gap: 1rem;
+        }
+    </style>
+</head>
+<body>
+
+<nav th:replace="~{fragments/profile_navbar}"></nav>
+
+<div class="container">
+    <div th:if="${notFound != null}" class="alert alert-warning" role="alert">
```
Comment (eugenesmkn) [inline src/main/resources/templates/public_page.html:62]: How do you use this block of code?

**Review by eugenesimakin — CHANGES_REQUESTED** [review]

**Comment by Washubrnfg85** [comment]
It works for me now.

**Comment by eugenesmkn** [comment]
User's public page which is open to the Internet should not contain this (when a user is not logged in there is no point in this):
image

And this:
image

Review your idea of understanding the app.

**Comment by Washubrnfg85** [comment]
We never declared it.

**Comment by Washubrnfg85** [comment]
Well. Users of this app, can collect useful links.
They can share their link lists with other app users.

File: src/main/resources/templates/public_page.html [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        <p>
+            Look like there is no data associated with this user.
+            Please, contact the person who shared this with you.
+        </p>
+    </div>
+
+    <div th:if="${notFound == null}">
+        <div class="public-link">
+            <h6>Copy link to share your profile:</h6>
+            <span th:text="${'http://localhost:8080/public/' + username}"></span>
```
Comment (eugenesmkn) [inline src/main/resources/templates/public_page.html]: Why don't you make this a link? (`<a>`)

**Review by eugenesimakin — CHANGES_REQUESTED** [review]

**Comment by eugenesmkn** [comment]
> Well. Users of this app, can collect useful links. They can share their link lists with other app users.

They can share their link lists with ~~other app users~~ _anyone_.

File: src/main/resources/templates/public_page.html (hunk 2 above)
Comment (Washubrnfg85) [reply src/main/resources/templates/public_page.html]: No reason. I could have thought of it myself.

**Review by eugenesimakin — APPROVED** [review]
Look okay to me. Can be merged.

**Pull request merged**